src/Typedefs/Test/TermParseWriteTests.idr

```diff
@@ -19,6 +19,27 @@ roundtrip1 td x = deserializeBinaryClosed td $ serializeBinaryClosed td x
 roundtrip2 : (td : TDefR 0) -> Bytes -> Maybe Bytes
 roundtrip2 td b = map (serializeBinaryClosed td . fst) (deserializeBinaryClosed td b)
 
+TNat : TDefR 0
+TNat = TMu [("Z", T1), ("S", TVar 0)]
+
+TList : TDefR 1
+TList = TMu [("Nil", T1), ("Cons", TProd [TVar 1, TVar 0])]
+
+toNat : Ty [] TNat -> Nat
+toNat (Inn (Left ()))   = Z
+toNat (Inn (Right inn)) = S $ toNat inn
+
+fromNat : Nat -> Ty [] TNat
+fromNat  Z    = Inn (Left ())
+fromNat (S n) = Inn (Right $ fromNat n)
+
+toList : Ty [] (TList `ap` [tdef]) -> List (Ty [] tdef)
+toList (Inn (Left ()))        = Nil
+toList (Inn (Right (hd, tl))) = ignoreShift hd :: toList tl
+
+fromList : List (Ty [] tdef) -> Ty [] (TList `ap` [tdef])
+fromList  Nil      = Inn (Left ())
+fromList (x :: xs) = Inn (Right (addShift x, fromList xs))
```

Comment:
Shouldn't this be somewhere else? It would be nice if we could expose it in as a "typedefs standard library"

Reply:
Ok, where do you think we should put this? We already have `examples` which cover both of these, but they're in a different subproject.

Reply:
How about : `src/Typedefs/Library`? 
we could have `Library.idr` be the standard library and subfolders be more specific imports like
`src/Typedefs/Library/JSON.idr` be all the functions related to using the JSON serialisation format

Reply:
Ok, should I move most of `examples` there too?

Reply:
I'm thinking about `bit`, `byte`, `maybe2` + helpers, `either2` + helpers, `list` and `nat`

Reply:
mhh probably just Maybe and the generic stuff, `bit` and `byte` doesn't really have a practical purpose do they? Maybe have `Bool` instead of `bit`